Wikipedia/Code/WMFAppViewController.m

```diff
@@ -716,6 +718,12 @@ - (void)runDataMigrationIfNeededWithCompletion:(dispatch_block_t)completion {
 #pragma mark - UITabBarControllerDelegate
 
 - (void)tabBarController:(UITabBarController*)tabBarController didSelectViewController:(UIViewController*)viewController {
+    NSInteger prevActiveTabIndex = [[NSUserDefaults standardUserDefaults] integerForKey:PrevActiveTabIndex];
+    [[NSUserDefaults standardUserDefaults] setInteger:tabBarController.selectedIndex forKey:PrevActiveTabIndex];
```

Comment:
Instead of saving this to user defaults, an alternate solution that doesn't require persistence would be to also implement:

```
- (BOOL)tabBarController:(UITabBarController *)tabBarController
shouldSelectViewController:(UIViewController *)viewController
```
https://developer.apple.com/library/ios/documentation/UIKit/Reference/UITabBarControllerDelegate_Protocol/index.html#//apple_ref/occ/intfm/UITabBarControllerDelegate/tabBarController:shouldSelectViewController:

You could capture the currently selected tab there, and if it is thew explore VC then you can scroll to the top - so that avoids using NSUserDefaults all together.